### ISSUE TITLE
Enhance Post Mechanism

### ISSUE THREAD

**Issue by ahmetfirat23** [bounswe/bounswe2024group12#309]
## Description

Develop post mechanism according to requirements.

***

## Acceptance Criteria
- [x] Posts should be likeable, dislikeable, shareable, deleteable.
- [x] Posts should be commentable. Comments should include everything posts has.
- [x] Extended comment page should be displayed when display more clicked.
<br>

***

### Reviewer:
This issue will be reviewed by: _@isilsukarakuzuu_

**Comment by ahmetfirat23** [comment]
I gave up on dislike because it was unnecessary to duplicate work. Sharing just copies the link of the post. Comments don't have everything posts have because the endpoint only has text and fen. I implemented the comment page, but instead of displaying more clicks, it is displayed when the comment button is clicked. Moreover, the endpoint should return the comment count. Also, endpoint should return whether the user liked the post or not. @sonerkuyar @yusufaygun
